hazelcast/src/main/java/com/hazelcast/collection/impl/list/ListProxyImpl.java

```diff
@@ -130,15 +132,15 @@ public boolean addAll(int index, Collection<? extends E> c) {
 
     @Override
     public List<E> subList(int fromIndex, int toIndex) {
-        final ListSubOperation operation = new ListSubOperation(name, fromIndex, toIndex);
-        final SerializableCollection result = invoke(operation);
-        final Collection<Data> collection = result.getCollection();
-        final List<E> list = new ArrayList<E>(collection.size());
+        ListSubOperation operation = new ListSubOperation(name, fromIndex, toIndex);
+        SerializableCollection result = invoke(operation);
+        Collection<Data> collection = result.getCollection();
```

Comment:
Isn't this a bit wastful? For you get back a collection, then you copied it into an ArrayList.

This generates a lot of waste.